### PR TITLE
Support super property reference in object methods/accessors

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/parser/js/js-parser-expr.c [hunk 1]
```diff
@@ -1008,16 +1022,32 @@ parser_parse_object_literal (parser_context_t *context_p) /**< context */
...
+        if (has_super_env)
+        {
+          context_p->last_cbc_opcode = CBC_PUSH_TWO_LITERALS;
+          context_p->last_cbc.value = function_literal_index;
+          parser_emit_cbc_ext (context_p, CBC_EXT_OBJECT_LITERAL_SET_HOME_OBJECT_COMPUTED);
+          parser_emit_cbc_ext (context_p, is_getter ? CBC_EXT_SET_COMPUTED_GETTER
+                                                    : CBC_EXT_SET_COMPUTED_SETTER);
+        }
+        else
+#endif /* ENABLED (JERRY_ESNEXT) */
```
[the remark this answers is not in the document]
Comment (rerobika) [reply jerry-core/parser/js/js-parser-expr.c:1046]: That's a frequently used style nowdays.

File: jerry-core/parser/js/byte-code.c [hunk 2]
```diff
@@ -27,7 +27,7 @@ JERRY_STATIC_ASSERT ((sizeof (cbc_uint16_arguments_t) % sizeof (jmem_cpointer_t)
  */
 JERRY_STATIC_ASSERT (CBC_END == 238,
                      number_of_cbc_opcodes_changed);
-JERRY_STATIC_ASSERT (CBC_EXT_END == 128,
+JERRY_STATIC_ASSERT (CBC_EXT_END == 132,
```
Comment (zherczeg) [inline jerry-core/parser/js/byte-code.c:30]: Is JERRY_SNAPSHOT_VERSION increased?

File: jerry-core/parser/js/js-scanner.c [hunk 3]
```diff
@@ -202,6 +201,12 @@ scanner_scan_primary_expression (parser_context_t *context_p, /**< context */
       break;
     }
 #if ENABLED (JERRY_ESNEXT)
+    case LEXER_KEYW_SUPER:
```
Comment (zherczeg) [inline jerry-core/parser/js/js-scanner.c:204]: Put this before LEXER_KEYW_THIS and use fallthrough.

File: jerry-core/parser/js/js-scanner.c [hunk 4]
```diff
@@ -586,14 +593,17 @@ scanner_scan_primary_expression_end (parser_context_t *context_p, /**< context *
...
                     || context_p->stack_top_uint8 == SCAN_STACK_FOR_LET_START
                     || context_p->stack_top_uint8 == SCAN_STACK_FOR_CONST_START
                     || context_p->stack_top_uint8 == SCAN_STACK_FUNCTION_PARAMETERS
                     || context_p->stack_top_uint8 == SCAN_STACK_ARROW_ARGUMENTS);
 
-      JERRY_ASSERT ((stack_top != SCAN_STACK_ARRAY_LITERAL && stack_top != SCAN_STACK_OBJECT_LITERAL)
+      JERRY_ASSERT ((stack_top != SCAN_STACK_ARRAY_LITERAL
+                     && stack_top != SCAN_STACK_OBJECT_LITERAL
+                     && stack_top != SCAN_STACK_OBJECT_LITERAL_WITH_SUPER)
                     || SCANNER_NEEDS_BINDING_LIST (scanner_context_p->binding_type));
```
Comment (zherczeg) [inline jerry-core/parser/js/js-scanner.c:607]: Hard to see. Can we move SCANNER_NEEDS_BINDING_LIST as the first condition?

File: jerry-core/parser/js/js-scanner.c (hunk 3 above)
Comment (rerobika) [reply jerry-core/parser/js/js-scanner.c:204]: There is already a fallthru for LEXER_KEYW_THIS.

**Pull request merged**